### ISSUE TITLE
Use crowdin for translations

### ISSUE THREAD

**Issue by ElDavoo** [Syncplay/syncplay#503]
That should make translating better

**Comment by Et0h** [comment]
Hi, thanks for sharing your suggestion.

I think it's great for Syncplay to be available in multiple languages and so I looked into CrowdIn a few years ago. I wasn't convinced CrowdIn was an improvement over our current approach given our use case. It did however appear to add to the complexity of the Syncplay code base and looked like it would require significant time and resources to implement and maintain.

As such I'd only be willing to consider it if an actual translator and coder with CrowdIn experience vouched for it being a significant improvement for our use case and volunteered to take the lead on such a project.

**Comment by comradekingu** [comment]
@Et0h You want to stay far away from Crowdin and use Weblate.
I can help if it the project uses one of https://docs.weblate.org/en/latest/formats.html?highlight=formats#translation-types-capabilities :)

**Comment by ElDavoo** [comment]
I guess I should have said "let's use a translation platform". I only know crowdin so I said crowdin :)

**Comment by Et0h** [comment]
@nipkownix @lryanner @sosie-js @OlegSmelov @muaz742 @artur309 @AtilioA @v-ji @bcattaneo @remimarenco @DerGenaue @and-semakin 

Do any of Syncplay's current/prospective translators have a view on whether moving to Weblate would make translations easier/harder for you?

**Comment by OlegSmelov** [comment]
I've never used a translation platform, but I think editing Python files can't be the most convenient way to translate. 😄 A translation platform that's free to use for open source projects is worth giving a shot IMO.

But in the end I think the decision is up to current maintainers if they feel like refactoring current code and integrating a translation platform is worth the effort.

**Comment by nipkownix** [comment]
I've only had one experience with a translation platform and it wasn't the most pleasant one. I'm very comfortable doing it the way we usually do, but I'd be open to giving a translation platform another shot if the maintainers think it is worth it.

**Comment by AtilioA** [comment]
I've translated in the past using both Crowdin and Weblate and I can say I prefer Weblate.
Unfortunately, I have no knowledge of how to implement it, but I'm afraid that it would not be worth the effort. Although the current process is not ideal for non-technical people, it can be easy enough to follow through. Also, since this is not a new project, most strings are already translated.

**Comment by bcattaneo** [comment]
> @nipkownix @lryanner @sosie-js @OlegSmelov @muaz742 @artur309 @AtilioA @v-ji @bcattaneo @remimarenco @DerGenaue @and-semakin @albertosottile
> 
> Do any of Syncplay's current/prospective translators have a view on whether moving to Weblate would make translations easier/harder for you?

Haven't used any of that, but I wouldn't mind some utility for translation. Even though it's not critical, it's still an improvement over the current method.

**Comment by artur309** [comment]
I'm just a regular newbie translator, anything new and great to use, im here to learn.

**Comment by muaz742** [comment]
I agree with the concern about outsourcing dependency. It's nice that translating is simple and functional. Maybe 'yaml' or 'csv' file types can be used to make it easier to use. (Python indentations are very useful when writing and reading code. I can't say the same when translating.)

Also, the translation tools on translation platforms are interesting.

I wonder if we write an API to control our data exchange with an external source, can we remove the workload and risks? Or can we include these tools in the translation system here?

**Comment by Et0h** [comment]
Thanks for the offer of support @comradekingu - we've never used Weblate, but it seems like if we were going down this route it would be a reasonable choice.

We are not currently using any of the systems listed at https://docs.weblate.org/en/latest/formats.html

Instead, our messages are spread across files such as https://github.com/Syncplay/syncplay/blob/master/syncplay/messages_en.py and across https://github.com/Syncplay/syncplay/blob/master/buildPy2exe.py

As such, before we could start moving towards Weblate we'd need to put time into changing how the NSIS file is generated and how the message system works and to actually convert Syncplay into one of the formats you mention. It's not something I can see myself having the time to take the lead on in the foreseeable future.

Maybe a more easily deliverable goal would be a guide in the Syncplay wiki to explain how people can make translations using our current system to help try to lower the barriers to entry. If anyone wanted to help start that off then this would be appreciated.
